liquibase-service/src/main/resources/db/rdb/023-sp_nrt_investigation_postprocessing-001.sql

```diff
@@ -509,14 +501,44 @@ BEGIN
                           from dbo.confirmation_method cd
                           where cd.confirmation_method_cd = cmt.confirmation_method_cd);
 
+        /* Logging */
+        set @rowcount = @@rowcount
+        INSERT INTO [dbo].[job_flow_log]
+        (batch_id
+        ,[Dataflow_Name]
+        ,[package_Name]
+        ,[Status_Type]
+        ,[step_number]
+        ,[step_name]
+        ,[row_count]
+        ,[msg_description1])
+        VALUES (@batch_id
+               ,@dataflow_name
+               ,@package_name
+               ,'START'
+               ,@proc_step_no
+               ,@proc_step_name
+               ,@rowcount
+               ,LEFT(@id_list, 500));
+
+        SET @proc_step_name = 'UPDATE CONFIRMATION_METHOD_GROUP';
+        SET @proc_step_no = 5;
+
+        delete dbo.CONFIRMATION_METHOD_GROUP
+        where investigation_key in
```

Comment:
Are there any cases or checks needed to confirm that these are the UIDs that need to be deleted?
Eg. If we get a list of 2 updates, 2 insert uids- of which 1 insert uid matches the requirement for delete and update, are we making sure we aren't deleting all the keys that are part of the update and regular insert?

Reply:
This is referencing Investigation directly

Reply:
I think there is no way to delete investigation, it can be only marked as inactive? And because sp does not have a block for investigation delete (insert/update only) we're safe here.

Reply:
Ah I meant for confirmation_method_group's delete statement. It uses only Investigation table as a reference. 
Would this delete investigation keys that came in as an update uid in the @id_list?

Reply:
It will delete the records containing investigation keys related to uids from @id_list, and the immediately insert new keys (investigation and confirmation method) related to payload that came for update. We always have confirmation method values in each upcoming investigation update. If there are no values - it means, that user unselected all the codes (=deleted confirmation method). So the procedure deletes all the codes (keys) related to the investigation and does not insert anything, because this is what user wanted.

Reply:
Got it! Thank you for explaining 💯  That makes sense.